### ISSUE TITLE
Crash when the destination of a desktop link is missing

### ISSUE THREAD

**Issue by tsujan** [lxde/pcmanfm-qt#592]
##### Steps to Reproduce the crash

1. Create a desktop link with a nonexistent destination:
```
[Desktop Entry]
Version=1.0
Encoding=UTF-8
Name=ANY_NAME
Type=Link
URL=PATH_TO_NONEXISTENT_FILE
TryExec=
Icon=ANY_ICON
...
```
2. Click on it in pcmanfm-qt to launch it.

A dialog will pop up stating that there's no such file or directory but pcmanfm-qt will crash after closing the dialog:
```
#0  0x00007f7594592a33 in fm_file_info_ref () at /usr/lib/libfm.so.4
#1  0x00007f7594594e12 in  () at /usr/lib/libfm.so.4
#2  0x00007f759459523c in fm_launch_files () at /usr/lib/libfm.so.4
#3  0x00007f7594595a37 in fm_launch_paths () at /usr/lib/libfm.so.4
#4  0x00007f759610704a in Fm::FileLauncher::launchPaths(QWidget*, _GList*) () at /usr/lib/libfm-qt.so.3
#5  0x00007f75961070b8 in Fm::FileLauncher::launchPaths(QWidget*, std::vector<Fm::FilePath, std::allocator<Fm::FilePath> >) () at /usr/lib/libfm-qt.so.3
#6  0x00007f75961071ee in Fm::FileLauncher::launchFiles(QWidget*, Fm::FileInfoList) () at /usr/lib/libfm-qt.so.3
#7  0x00007f7596116122 in Fm::FolderView::onFileClicked(int, std::shared_ptr<Fm::FileInfo const> const&) ()
    at /usr/lib/libfm-qt.so.3
#8  0x000000b9048ba80b in  ()
#9  0x00007f7594a7f8af in QMetaObject::activate(QObject*, int, int, void**) () at /usr/lib/libQt5Core.so.5
#10 0x00007f7596168373 in Fm::FolderView::clicked(int, std::shared_ptr<Fm::FileInfo const> const&) ()
    at /usr/lib/libfm-qt.so.3
#11 0x00007f7596111727 in Fm::FolderView::onItemActivated(QModelIndex) () at /usr/lib/libfm-qt.so.3
#12 0x00007f7596118c8d in  () at /usr/lib/libfm-qt.so.3
#13 0x00007f7594a7f8af in QMetaObject::activate(QObject*, int, int, void**) () at /usr/lib/libQt5Core.so.5
#14 0x00007f7596168442 in  () at /usr/lib/libfm-qt.so.3
#15 0x00007f7594a7f8af in QMetaObject::activate(QObject*, int, int, void**) () at /usr/lib/libQt5Core.so.5
#16 0x00007f75959c9306 in QAbstractItemView::activated(QModelIndex const&) () at /usr/lib/libQt5Widgets.so.5
#17 0x00007f75959cc49d in QAbstractItemView::mouseReleaseEvent(QMouseEvent*) () at /usr/lib/libQt5Widgets.so.5
#18 0x00007f75959f3c90 in QListView::mouseReleaseEvent(QMouseEvent*) () at /usr/lib/libQt5Widgets.so.5
#19 0x00007f7596117997 in  () at /usr/lib/libfm-qt.so.3
#20 0x00007f75957b8d68 in QWidget::event(QEvent*) () at /usr/lib/libQt5Widgets.so.5
#21 0x00007f759589c60f in QFrame::event(QEvent*) () at /usr/lib/libQt5Widgets.so.5
#22 0x00007f75959d54a4 in QAbstractItemView::viewportEvent(QEvent*) () at /usr/lib/libQt5Widgets.so.5
#23 0x00007f7594a4f89c in QCoreApplicationPrivate::sendThroughObjectEventFilters(QObject*, QEvent*) ()
    at /usr/lib/libQt5Core.so.5
#24 0x00007f7595776e28 in QApplicationPrivate::notify_helper(QObject*, QEvent*) () at /usr/lib/libQt5Widgets.so.5
#25 0x00007f759577eea1 in QApplication::notify(QObject*, QEvent*) () at /usr/lib/libQt5Widgets.so.5
#26 0x00007f7594a4fbd0 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () at /usr/lib/libQt5Core.so.5
#27 0x00007f759577de2a in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool) () at /usr/lib/libQt5Widgets.so.5
#28 0x00007f75957d3f51 in  () at /usr/lib/libQt5Widgets.so.5
#29 0x00007f75957d6691 in  () at /usr/lib/libQt5Widgets.so.5
#30 0x00007f7595776e4c in QApplicationPrivate::notify_helper(QObject*, QEvent*) () at /usr/lib/libQt5Widgets.so.5
#31 0x00007f759577e926 in QApplication::notify(QObject*, QEvent*) () at /usr/lib/libQt5Widgets.so.5
#32 0x00007f7594a4fbd0 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () at /usr/lib/libQt5Core.so.5
#33 0x00007f7594fb21e4 in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) ()
    at /usr/lib/libQt5Gui.so.5
#34 0x00007f7594fb3d46 in QGuiApplicationPrivate::processWindowSystemEvent(QWindowSystemInterfacePrivate::WindowSystemEvent*) () at /usr/lib/libQt5Gui.so.5
#35 0x00007f7594f8b4ac in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) ()
    at /usr/lib/libQt5Gui.so.5
#36 0x00007f758b7f5cc1 in  () at /usr/lib/libQt5XcbQpa.so.5
#37 0x00007f7593cc4a57 in g_main_context_dispatch () at /usr/lib/libglib-2.0.so.0
#38 0x00007f7593cc4c88 in  () at /usr/lib/libglib-2.0.so.0
#39 0x00007f7593cc4d1c in g_main_context_iteration () at /usr/lib/libglib-2.0.so.0
#40 0x00007f7594aaa061 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) ()
    at /usr/lib/libQt5Core.so.5
#41 0x00007f7594a4dffb in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () at /usr/lib/libQt5Core.so.5
#42 0x00007f7594a57048 in QCoreApplication::exec() () at /usr/lib/libQt5Core.so.5
#43 0x000000b904899113 in  ()
#44 0x00007f75933434ca in __libc_start_main () at /usr/lib/libc.so.6
#45 0x000000b90489917a in _start ()
```

##### System Information
Everything git up-to-date.

**Comment by tsujan** [comment]
@agaida, please start the rescue mission whenever you have time: https://github.com/lxde/libfm/pull/34

**Comment by tsujan** [comment]
@agaida And please show a little more "LXQt selfishness" and also merge https://github.com/lxde/libfm/pull/21 -- after testing and seeing that it works as intended, of course. It fixes a real security issue (https://github.com/lxde/pcmanfm-qt/issues/449), which was discussed a long time ago. I assure you that it won't be reviewed if you don't so so.

**Comment by agaida** [comment]
And it would fix it for LXDE too? I have a problem with it - sure i can merge it, but i can't upload it to debian.

**Comment by tsujan** [comment]
> And it would fix it for LXDE too?

Should do so but I don't have LXDE.

I take full responsibility for it. Like my other patches, I've used it since the PR. It's really bad to click on a desktop file and see it executed without a prompt.

**Comment by agaida** [comment]
:D - Never had any doubt. But in this case i think it is only fair to involve @LStranger and roll it out as soon as possible.

**Comment by tsujan** [comment]
> i think it is only fair to involve @LStranger and roll it out as soon as possible.

I don't think so; it'll mean *never*. Sorry that I'm so frank.

**Comment by tsujan** [comment]
Frankly, I've had this problem with @LStranger from start. The unattended list of libfm PR's shows that I'm not alone. For me, enough is enough.

**Comment by agaida** [comment]
Merging it is one thing - bringing it to the users another - and i guess @gilir and @tsimonq2 should also be notified.

**Comment by tsujan** [comment]
> Merging it is one thing - bringing it to the users another

You're the expert in both cases.

**Comment by LStranger** [comment]
>> i think it is only fair to involve @LStranger and roll it out as soon as possible.

>I don't think so; it'll mean *never*. Sorry that I'm so frank.

Well, if there is a fix for that then why not to fix it ages ago? And I'm
not only one developer of it who can push the merge button, you know, so
why should I be blamed for everything? Don't get me wrong, please, I have
my personal life too and thinking I have to solve everything isn't right.
I'm sorry that I could not review all PRs and make a release earlier due
to my personal life and due to crash of my home comp disk but as I said,
someone else could do it long ago as well. After all, if some PR would
create a side effect problem, it can be fixed by another fix, as it have
to be done in development usually and it always been. All of my earlier
comments were about stable release of 1.2.x series, which is anyway just
a side branch, the same way as it often happens in production cycle of
commercial software for its stable releases.

So TL;DR - feel free to merge any PRs you find working well and don't
blame me for never merging them, please, that way is going to nowhere.

**Comment by tsujan** [comment]
@LStranger My last response to such words: All of us have personal lives and all of us have life problems -- including me. My problems *may* be more than yours; I just don't talk about them: food for thought.

**Comment by tsimonq2** [comment]
> Merging it is one thing - bringing it to the users another - and i guess @gilir and @tsimonq2 should also be notified.

Ack, I'll assign this to myself to get fixed in Ubuntu.

**Comment by tsujan** [comment]
By chance, I found that AOSC-Dev uses https://github.com/lxde/libfm/pull/21 -- see [this](https://github.com/AOSC-Dev/aosc-os-abbs/commit/1eb67e4c731972207f805a92d0ac34b7c157cb20)!

But *we* still don't!!

**Comment by tsujan** [comment]
If https://github.com/lxde/libfm/pull/21 is merged, I might add a small libfm-qt PR, although it isn't necessary.

**Comment by LStranger** [comment]
@tsujan, I'm sorry to mention that but my text was not about personal
life but about the fact I'm not only one who could push merge button.

**Comment by tsujan** [comment]
Pushed it myself.
